### ISSUE TITLE
Fix the handling of the fiducial area cut in ixpeobssim

### ISSUE THREAD

**Issue by lucabaldini** [lucabaldini/ixpeobssim#668]
This is an offspring of issue #663 and I am opening it to keep track of the various ramifications of the changes we are doing in the code handling the fiducial area cuts. 

In a nutshell, we wan to move away from the old situation, where the fiducial region on the GPD surface was a square of 7.35 mm half-side, hard coded, and default to the new one, where the fiducial area is a rectangle whose half side are 6.6 and 6.8 mm on the x and y coordinates, respectively.\

Additionally, since when processing flight data at the SOC we haven't always consistent in applying the intended fiducial cut, we want to be able to change the fiducial region from command line---although this is a lower-priority requirement.

**Comment by lucabaldini** [comment]
Getting started on this. We have 

```
def gpd_map_binning(num_bins):
    """Return a numpy array with an appropriate binning for a bidimensional
    map over the GPD area.
    """
    return numpy.linspace(-FIDUCIAL_HALF_SIZE, FIDUCIAL_HALF_SIZE, num_bins + 1)
```

that needs to be adapted to a generic rectangle.

And the occurrences are:

```
[lbaldini@nblbaldini ixpeobssim]$ grep gpd_map_binning -r .
./ixpeobssim/binning/detector.py:from ixpeobssim.instrument.gpd import gpd_map_binning
./ixpeobssim/binning/detector.py:        binning = gpd_map_binning(self.get('npix'))
grep: ./ixpeobssim/binning/__pycache__/detector.cpython-310.pyc: binary file matches
./ixpeobssim/instrument/gpd.py:def gpd_map_binning(num_bins):
grep: ./ixpeobssim/instrument/__pycache__/gpd.cpython-310.pyc: binary file matches
./ixpeobssim/srcmodel/calibsrc.py:from ixpeobssim.instrument.gpd import gpd_map_binning, FIDUCIAL_HALF_SIZE
./ixpeobssim/srcmodel/calibsrc.py:        self.xbinning = gpd_map_binning(self.shape[0])
./ixpeobssim/srcmodel/calibsrc.py:        self.ybinning = gpd_map_binning(self.shape[1])
```

**Comment by lucabaldini** [comment]
Relevant commit for the gpd_map_binning refactoring:
https://github.com/lucabaldini/ixpeobssim/commit/6d94e8c68bbc6e7e538ef969a16bd2b022fba995

**Comment by lucabaldini** [comment]
The next thing to figure out is whether we want default values in the within_fiducial_area() hook.

```
[lbaldini@nblbaldini ixpeobssim]$ grep within_fiducial_area -r .
./ixpeobssim/binning/exposure.py:from ixpeobssim.instrument.gpd import within_fiducial_area
./ixpeobssim/binning/exposure.py:            mask = (z != thetabins) * (within_fiducial_area(detx, dety))
./ixpeobssim/evt/event.py:from ixpeobssim.instrument.gpd import within_fiducial_area
./ixpeobssim/evt/event.py:        mask = within_fiducial_area(detx, dety)
./ixpeobssim/instrument/gpd.py:def within_fiducial_area(x, y, half_side_x=GPD_DEFAULT_FIDUCIAL_HALF_SIDE_X,
```

**Comment by lucabaldini** [comment]
Commit for the within_fiducial_area() function: https://github.com/lucabaldini/ixpeobssim/commit/0f510d4e64ac057c61274c19a4739daa3150c78d

**Comment by lucabaldini** [comment]
Ok, so this has interesting ramifications---definitely more than I expected.

The instrumental background is currently sampled over the (old) fiducial square, which is both (slightly) wrong due to the fact that the entire surface contributes to the background and inconvenient in the moment where we decide that the fiducial cut is no more set ins stone.

I think the way to go is to generate the background over the entire GPD active area, and then trim to the target fiducial rectangle.

For most of the other thing we can probably get away by replacing the fiducial size with the physical size of the GPD, although we need to pay attention to the charging bits.

**Comment by lucabaldini** [comment]
Relevant commit for the GPD maps and xpstokessmear: https://github.com/lucabaldini/ixpeobssim/commit/8f78e21aa17a53bb8f59059491f74866aed2aa59

**Comment by lucabaldini** [comment]
Ok, I made some more minor edits. The MMA stuff is interesting:

```
FOCAL_LENGTH = 3997.
FIELD_OF_VIEW = numpy.degrees(2. * FIDUCIAL_HALF_SIZE / FOCAL_LENGTH)
FIDUCIAL_BACKSCAL = degrees_to_arcsec(FIELD_OF_VIEW)**2.
```

The only thing that apparently is used is the FIDUCIAL_BACKSCALE.

**Comment by lucabaldini** [comment]
Fiducial backscale refactoring: https://github.com/lucabaldini/ixpeobssim/commit/51f41e80dafc6e9876f8aeb15a120fc6ff535c16

**Comment by lucabaldini** [comment]
Shipped in 29.5.0
